### PR TITLE
teleport_15: init at 15.2.2, teleport_12: remove 

### PR DESCRIPTION
**Pull request by justinas**
## Description of changes

Adding [Teleport 15](https://github.com/gravitational/teleport/releases/tag/v15.0.0), a new major version.

Removing Teleport 12, because it is effectively EOL as the upstream only officially supports the 3 latest major versions at any given time.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/teleport/generic.nix [hunk 1]
```diff
@@ -154,7 +182,7 @@ buildGoModule rec {
   meta = with lib; {
     description = "Certificate authority and access plane for SSH, Kubernetes, web applications, and databases";
     homepage = "https://goteleport.com/";
-    license = licenses.asl20;
+    license = if lib.versionAtLeast version "15" then licenses.agpl3Plus else licenses.asl20;
```
Comment (justinas) [inline pkgs/servers/teleport/generic.nix:185]: Teleport changed the license to AGPLv3 (or later) staring with version 15. https://goteleport.com/blog/teleport-oss-switches-to-agpl-v3/

**Comment by justinas** [comment]
I am not sure how to make the `by-name` check happy given multiple versions living in `teleport/14` `teleport/15` etc.

**Comment by infinisil** [comment]
The `pkgs/by-name` only requires _new_ packages to use it, and that is based on whether the new definition looks like `<attr> = callPackage ...`, which is the case here.

Since there's a `generic.nix` file used by all the versions, you're a bit [more restricted](https://github.com/NixOS/nixpkgs/tree/master/pkgs/test/nixpkgs-check-by-name#nix-parser-checks) in what you can do, but there's some options:
- My recommendation: Avoid `pkgs/by-name` for now by creating a package set for all versions, either like this:
  ```nix
    teleportVersions = callPackages ../servers/teleport { };
    inherit (teleportVersions) teleport_14 teleport_15 teleport;
  ```

  Or if you don't want to expose `teleportVersions` to users:
  ```nix
    inherit (callPackages ../servers/teleport { })
      teleport_14 teleport_15 teleport;
  ```
  - There's a bug in the `pkgs/by-name` check that would still complain about this, see https://github.com/NixOS/nixpkgs/pull/285089, but this would be allowed
- Alternatively, expose `generic.nix` as `pkgs.teleportGeneric`, such that you can refactor the individual versions to use `pkgs/by-name`, each of which taking `teleportGeneric` as an argument:
  ```nix
  { teleportGeneric }:
  teleportGeneric { ... }
  ```

**Comment by justinas** [comment]
Thanks for the advice @infinisil , I ended up utilizing `callPackages` and the check passes now.

File: pkgs/servers/teleport/generic.nix [hunk 2]
```diff
@@ -88,7 +108,15 @@ let
...
 
-      yarn build-ui-oss
+      ${if lib.versionAtLeast version "15"
+      then ''
+        PATH=$PATH:$PWD/node_modules/.bin
+        pushd web/packages/teleport
+        wasm-pack build ./src/ironrdp --target web --mode no-install
+        vite build
+        popd
+        ''
```
Comment (SuperSandro2000) [inline pkgs/servers/teleport/generic.nix:118]:
```suggestion
      ''
```

File: pkgs/servers/teleport/14/default.nix [hunk 3]
```diff
@@ -1,5 +1,5 @@
-{ callPackage, ... }@args:
-callPackage ../generic.nix ({
+{ ... }@args:
```
Comment (SuperSandro2000) [inline pkgs/servers/teleport/14/default.nix:1]:
```suggestion
args:
```

File: pkgs/servers/teleport/13/default.nix [hunk 4]
```diff
@@ -1,5 +1,5 @@
-{ callPackage, ... }@args:
-callPackage ../generic.nix ({
+{ ... }@args:
```
Comment (SuperSandro2000) [inline pkgs/servers/teleport/13/default.nix:1]:
```suggestion
args:
```

**Comment by SuperSandro2000** [comment]
Can you rebase? otherwise LGTM

**Comment by justinas** [comment]
Rebased.

**Review by techknowlogick — APPROVED** [review]

**Review by tomberek — APPROVED** [review]

**Pull request merged**